rbac/api/cross_access/view.py

```diff
@@ -107,6 +112,52 @@ def list(self, request, *args, **kwargs):
             return self.replace_user_id_with_info(result)
         return result
 
+    def partial_update(self, request, *args, **kwargs):
+        """Patch a cross-account request."""
+        validate_uuid(kwargs.get("pk"), "cross-account request uuid validation")
+        for field in request.data:
+            if field not in VALID_PATCH_FIELDS:
+                self.throw_validation_error(
+                    "cross-accont partial update",
+                    f"Field '{field}' is not supported. Please use one or more of: {VALID_PATCH_FIELDS}"
+                )
+
+        with tenant_context(Tenant.objects.get(schema_name="public")):
+            current = self.get_object()
+
+            if current.status != "pending":
+                self.throw_validation_error("cross-account partial update", "Only pending requests may be updated.")
+
+            self.validate_and_format_input(request.data)
+
+            kwargs["partial"] = True
+            response = super().update(request=request, *args, **kwargs)
+            if response.status_code and response.status_code is http_status.HTTP_200_OK:
+                if request.data.get("status"):
+                    self.update_status(current, request.data.get("status"))
+                    return Response(CrossAccountRequestDetailSerializer(current).data)
+            return response
+
+    def update(self, request, *args, **kwargs):
+        """Update a cross-account request."""
+        validate_uuid(kwargs.get("pk"), "cross-account request uuid validation")
+
+        with tenant_context(Tenant.objects.get(schema_name="public")):
+            current = self.get_object()
+            if current.status != "pending":
+                self.throw_validation_error("cross-account update", "Only pending requests may be updated.")
+            if "target_account" in request.data and str(request.data.get("target_account")) != current.target_account:
+                self.throw_validation_error("cross-account-update", "Target account may not be updated.")
+
+            self.validate_and_format_input(request.data)
```

Comment:
I think I still have a question here around being able to update the `target_account`. If you supply it in a `PUT`, you can override it currently, correct? I feel like we may want to restrict that similar to the `PATCH`?

Reply:
Ah, line 149 may handle this now?

Reply:
Yep! You can supply `target_account` since it's a required field, but it has to be the same as originally created with. I could alternatively just grab the one from the current object and shim that in if we'd rather.